src/services/userService.ts

```diff
@@ -1,21 +1,108 @@
-import { LoginCredentials } from "../domain/LoginCredentials";
-import Success from "../domain/Success";
 import logger from "../misc/logger";
-import * as userModel from "../models/userModels";
+import Success from "../domain/Success";
+import userModel from "../models/userModels";
+import User, { UserToInsert } from "../domain/User";
+import bcrypt from "bcrypt";
+
+import Token from "../domain/Token";
+
+import jwt from "jsonwebtoken";
+
+export const login = async (
+    email: string,
+    password: string
+): Promise<Success<Token>> => {
+    const user = await userModel.getUserByEmail(email);
+    if (!user) {
+        return {
+            message: "Invalid email or password",
+        };
+    }
+    const isPasswordMatch = await bcrypt.compare(password, user.password);
+    if (!isPasswordMatch) {
+        return {
+            message: "Password does not match",
+        };
+    }
+
+    const accessToken = jwt.sign(
+        { userId: user.id },
+        process.env.JWT_SECRET as string
+    );
+
+    return {
+        data: { accessToken },
+        message: "User logged in successfully",
+    };
+};
 
-/**
- * Get user login credentials
- * @returns {Promise<LoginCredentials>}
- */
-export const getUserLoginCredentials = async (
-    userName: string
-): Promise<Success<LoginCredentials>> => {
-    logger.info(`Getting login credentials with username: ${userName}`);
+export const getAllUsers = async (): Promise<Success<User[]>> => {
+    logger.info("Getting all users");
 
-    const user = await userModel.getUserLoginCredentials(userName);
+    const users = await userModel.getAllUsers();
+
+    return {
+        data: users,
+        message: "Users fetched successfully",
```

Comment:
Does this data return reveal the passwords as well? We would want password to be not seen in response.

Reply:
Ok, i will fix that